### ISSUE TITLE
Compiling p4p as EPICS module on windows

### ISSUE THREAD

**Issue by beenje** [mdavidsaver/p4p#47]
Is compiling p4p as EPICS module on windows supported?
I get the following error when trying:
```
make -C ./configure install
make[1]: Entering directory 'C:/Users/IEUser/Dev/p4p/configure'
perl -CSD C:/epics/bin/windows-x64/makeMakefile.pl O.windows-x64 ../..
perl -CSD -MExtUtils::Command -e mkpath O.Common
make -C O.windows-x64 -f ../Makefile TOP=../.. \
    T_A=windows-x64 install
make[2]: Entering directory 'C:/Users/IEUser/Dev/p4p/configure/O.windows-x64'
[ -d os/ ] || perl -CSD -MExtUtils::Command -e mkpath os/
'[' is not recognized as an internal or external command,
operable program or batch file.
python ../../makehelper.py os/CONFIG_PY.Common.windows-x64
Traceback (most recent call last):
  File "../../makehelper.py", line 47, in <module>
    print('PY_LIBDIRS :=',' '.join(libdirs), file=out)
TypeError: sequence item 0: expected str instance, NoneType found
make[2]: *** [../Makefile:19: os/CONFIG_PY.Common.windows-x64] Error 1
make[2]: Leaving directory 'C:/Users/IEUser/Dev/p4p/configure/O.windows-x64'
make[1]: *** [C:/epics/configure/RULES_ARCHS:58: install.windows-x64] Error 2
make[1]: Leaving directory 'C:/Users/IEUser/Dev/p4p/configure'
make: *** [C:/epics/configure/RULES_DIRS:84: configure.install] Error 2
```

I'd like to compile p4p with my own epics-base (without using epicscorelibs).
It's working fine on Linux.

**Comment by mdavidsaver** [comment]
P4P is [tested on windows](https://ci.appveyor.com/project/mdavidsaver/p4p).  Though, as I look at it, only the setup.py build method.  I guess I have an inbuilt assumption that windows users would always prefer the pre-built wheel binaries.

**Comment by mdavidsaver** [comment]
Adding an appveyor build w/ Makefile is a non-trivial task which I don't have time for atm.  #48 is an attempt to address the specific errors reported so far (there will likely be more).

**Comment by hhslepicka** [comment]
@beenje the current `p4p` recipe on `conda-forge` uses epicscorelibs and there is no issue on using it along with `epics-base`. Do you mind testing it and report back?
As maintainer of `p4p` on `conda-forge` I don't intend to change the dependency to `epics-base` if it works.

**Comment by beenje** [comment]
@hhslepicka, I've used p4p installed with pip (and epicscorelibs) in a conda env with epics-base. It works.
I understand why epicscorelibs exists for pip. But with conda, I personally think it would make more sense to depend on the existing epics-base. I don't see why we need 2 different conda packages for epics libraries.
I was planning a PR when I get it working on all platforms. But it's up to you as maintainer :-)
At ESS I use this recipe: https://gitlab.esss.lu.se/conda-recipes/p4p-recipe (we build only for linux)

**Comment by mdavidsaver** [comment]
I assume this was fixed by #48.
